packages/vats/test/test-boot-config.js

```diff
@@ -18,34 +23,144 @@ const CONFIG_FILES = [
+  const runViz = pspawn(vizTool, { spawn: ambientSpawn });
+
+  return {
+    asset,
+    bundleCache,
+    cacheDir,
+    resolve,
+    basename: path.basename,
+    runViz,
+  };
+};
+
+test.before(async t => {
+  t.context = await makeTestContext();
 });
+// #endregion
 
-test('Bootstrap SwingSet config file syntax', /** @param {TestCtx} t */ async t => {
+test('Bootstrap SwingSet config file syntax', async t => {
   const { asset } = t.context;
 
   await Promise.all(
     CONFIG_FILES.map(async f => {
       const txt = await asset('..', f);
       const config = harden(JSON.parse(txt));
-      await t.notThrows(() => mustMatch(config, ssShape.SwingSetConfig), f);
+      t.notThrows(() => mustMatch(config, ssShape.SwingSetConfig), f);
       const parameters = config?.vats?.bootstrap?.parameters;
       t.log('syntax check:', f, parameters ? 'and parameters' : '');
-      (await parameters) &&
+      parameters &&
         t.notThrows(() => mustMatch(parameters, BootParametersShape), f);
     }),
   );
 });
+
+test('no test-only code is in production configs', async t => {
+  const { basename, bundleCache, resolve } = t.context;
+  const { entries } = Object;
+
+  const seen = new Set();
+
+  const noLog = () => {};
+
+  for await (const configSpec of PROD_CONFIG_FILES) {
+    t.log('checking config', configSpec);
+    const fullPath = resolve('..', configSpec);
+    const config = await loadSwingsetConfigFile(fullPath);
+    if (!config) throw t.truthy(config, configSpec); // if/throw refines type
+    const { bundles } = config;
+    if (!bundles) throw t.truthy(bundles, configSpec);
+
+    for await (const [name, spec] of entries(bundles)) {
+      if (!('sourceSpec' in spec)) throw t.fail();
+
+      // t.log('checking bundle path', name, spec.sourceSpec);
+      for (const vatName of NON_UPGRADEABLE_VATS) {
+        if (spec.sourceSpec.includes(vatName)) {
+          t.fail(`${configSpec} bundle ${spec.sourceSpec} not upgradeable`);
+        }
+      }
+
+      await bundleCache.load(spec.sourceSpec, undefined, noLog);
+      const targetName = basename(spec.sourceSpec, '.js');
+
+      if (!seen.has(targetName)) {
+        seen.add(targetName);
+        t.log('checking bundle', targetName);
+        const meta = await bundleCache.validate(targetName);
+        t.truthy(meta, name);
+
+        for (const item of meta.contents) {
+          if (item.relativePath.includes('magic-cookie-test-only')) {
+            t.fail(`${configSpec} bundle ${name}: ${item.relativePath}`);
+          }
+        }
+      }
+    }
+  }
+});
+
+test('bootstrap permit visualization snapshot', async t => {
+  const { runViz } = t.context;
+
+  const cmd = runViz(['@agoric/vats/decentral-test-vaults-config.json']);
+  const output = async () => {
+    const parts = [];
+    cmd.child.stdout?.on('data', chunk => parts.push(chunk));
+    await cmd.exit;
+    return parts.join('');
+  };
+  const diagram = await output();
+  t.snapshot(diagram);
```

Comment:
I wonder if committing to a specific set of bootstrap permits is overly specific. hm.

Reply:
fine for now. nice to have tests to keep this thing working.